### DISCUSSION TITLE
More 'smooth' auto mode

### DISCUSSION THREAD

**Discussion by HV-NL** [revk/xESP32-Faikout#130] [Q&A]
I intend to use the Faikin/Daikin combination to keep my room on temperature (i.e. heating). Target is 18 °C.

When I enable the 'auto' mode the 'Set' ruler goes to 27 °C. As a reaction the outdoor unit starts at almost max power (54 Hz). After some minutes 'Set' even moves to 30 °C. And suddenly 'Set' is set to 16 °C, effectively forcing the outdoor unit to halt completely. As an effect the indoor unit also stops. After some minutes this process is repeated, and so on, and so on (in dutch: 'pendelen').

What I want is a much smoother situation. So 'Set' not at 27 °C (and certainly not to 30 °C), but one or two degrees above the internal temperature of the unit., so that the outdoor unit runs at 20 Hz - 30 Hz.
And when the system thinks that the required temperature is met, not setting back to 16 °C, but to the required temperature (in my example 18 °C), in order to not completely stop the outdoor unit, but to slow it down (e.g. running at 15 Hz or 18 Hz or so, i.s.o. 0 Hz).

Are there any settings that I can adapt to reach this, and thus to prevent all these start/stop sequences, which will in the end result in a much more comfortable situation?

![afbeelding](https://github.com/revk/ESP32-Faikin/assets/33630544/2d0778be-2167-442a-ad21-c916fbd8cfae)

**Comment by revk** [comment]
There are several settings that can be controlled by MQTT, but the design works well with my air-con and can maintain temperature within a fraction of a degree. It does work by setting a much higher or lower temperature to get the air-con to  heat or stop heating, or cool or stop cooling, as appropriate. If that does not work for your unit then that may not be the best way to manage this, sorry. The issues is there is no direct way to control the compressor, we only really have temperature target as a control.

See https://github.com/revk/ESP32-Faikin/blob/main/Manuals/Advanced.md for more details of the settings that can be set via MQTT. If you find some that work better, do post details or add to the wiki. Thanks.

**Reply by HV-NL** [reply]
>  _It does work by setting a much higher or lower temperature_ 

I looked at the advanced page, but did not see a value that represents 'much higher'. My guess is that when the setting is not 'much higher' but 'a little higher' that the outdoor unit will react more the way I prefer.

Can you suggest what items I should configure? Should I try to reconfigure 'tmin' and 'tmax'?

**Reply by revk** [reply]
<img width="1044" alt="Screenshot 2023-10-17 at 17 36 04" src="https://github.com/revk/ESP32-Faikin/assets/996983/c5eb5e1f-2d3b-4b18-b665-3025e1026da8">

**Reply by HV-NL** [reply]
Thank you. 
Do you have an example how to send such a setting/heatover command using mosquitto_pub, or with any other commandline tool that can create MQTT messages on Raspberry Pi?

**Reply by revk** [reply]
A topic of `setting/nameofdevice/heatover` and payload of the value.

Also possible to send `setting/nameofdevice` and a JSON payload such as `{"heatover":5}`

**Comment by HV-NL** [comment]
> See https://github.com/revk/ESP32-Faikin/blob/main/Manuals/Advanced.md for more details of the settings that can be set via MQTT. If you find some that work better, do post details or add to the wiki. Thanks.

Currently I use the following settings:
"heatback": 0
"heatover": 0
"coolback": 0
"coolover": 0
This results in a much smoother behavior.

**Reply by revk** [reply]
Well done.

**Comment by turbolooser** [comment]
I am playing around with these settings also. Could you tell what the defaults are for these 4 parameters?

**Comment by turbolooser** [comment]
Just to add from my side.
These parameters defaults are "6" and do not require any change to get the hysterese more smooth on my daikin multisplit in auto mode. Setting the demand control to the lowest possible value to reach required Liquid temps from the heat exchanger does the trick for me. 

To me its like a combination of
- target temp needs to be a good amount higher (2-4 celsius) so that my perferas in combination with 3MXM40 continously produces enough heat
- fanspeed needs to be (medium/higher at least initially) capable of transporting the heat of the outside unit to the inside unit so that it can give it to the room (if fanspeeds are to loow then the outside unit will not produce higher temps)
- demand control as low as possible will make the outside unit continously run on lower power consumption rather than going high/low/off with required amount of heat for the inside units

Just wanna share :-)

**Reply by revk** [reply]
Thanks.

**Reply by macekkrystof** [reply]
This sounds interesting to me. Just got multisplit daikin installed and struggling to make it work smoothly. Premade Faikin PCB is sold out. Do you think I can achieve what you describe just with Onecta app (+Daikin controller in home assistant)? I tried dynamically changing AC target temperature (based on actual room temperature from Tado thermostats) via automations in Home assistant but still not smooth enough. Or is Faikin required to achieve this behavior. I can set demand control in onecta app.

**Reply by revk** [reply]
FYI pre-made Faikin will be in stock in a few days I expect.

**Reply by macekkrystof** [reply]
@revk That's nice! Just a bit worried about my warranty 😬

**Reply by revk** [reply]
Sadly I cannot help with that.

**Reply by macekkrystof** [reply]
I'll probably become too annoyed by the Onecta app that I'll buy it anyway :)

**Reply by turbolooser** [reply]
You are just using the S21 in addition to what Daikin is giving you as default. I am not worried about my warranty

**Reply by macekkrystof** [reply]
@turbolooser  just tried to set demand control like you recommended and it is much better. Like 40% lower power consumption and maintaining target temperature more smoothly.
